src/dependency-manager/src/manager.ts

```diff
@@ -98,10 +112,15 @@ export default abstract class DependencyManager {
       if (node instanceof ServiceNode) {
         for (const [param_name, param_value] of Object.entries(node.parameters)) { // load param references
           if (param_value instanceof Object && param_value.valueFrom) {
-            const param_target_service_name = (param_value as ValueFromParameter).valueFrom.dependency;
+            const value_from_param = param_value as ValueFromParameter;
+            const param_target_service_name = value_from_param.valueFrom.dependency;
             const param_target_datastore_name = (param_value as DatastoreValueFromParameter).valueFrom.datastore;
+
             if (param_target_service_name) {
-              const param_target_service = this.graph.getNodeByRef(param_target_service_name);
+              if (value_from_param.valueFrom.interface && !all_interface_names.includes(value_from_param.valueFrom.interface)) {
```

Comment:
This should only check for the interfaces on the param_target_service. 

`if (value_from_param.valueFrom.interface && value_from_param.valueFrom.interface in param_target_service.interfaces)`